.github/workflows/update-online-api-docs.yml

```diff
@@ -0,0 +1,80 @@
+name: Update Online API Docs
+
+on:
+  # The scheduled job for nightly API docs
+  schedule:
+    # Trigger at UTC 00:00 everyday
+    - cron: "0 0 * * *"
+  # The job for API docs of new version
+  push:
+    branches:
+      - main
+    paths:
+      - VERSION
+
+jobs:
+  build_and_upload:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    container: asterinas/asterinas:0.4.2
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          repository: 'asterinas/asterinas'
+          path: 'asterinas'
+
+      - name: Build & Upload nightly API docs
+        if: github.event_name == 'schedule'
+        env:
+          API_DOCS_NIGHTLY_PUBLISH_KEY: ${{ secrets.API_DOCS_NIGHTLY_PUBLISH_KEY }}
+        run: |
```

Comment:
The instructions for "Build & Upload nightly API docs" and "Build & Upload new-version API docs" looks quite alike, with only a few differences. Is it possible to extract the common needs to upload websites by adding a script under `tools/`? Using the script would simplify the workflow yml files.